src/AllBookings/BookingTypes/MulticityBooking.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+
+import idx from 'idx';
+import * as React from 'react';
+import { graphql, createFragmentContainer } from 'react-relay';
+
+import BookingCard from '../BookingCard';
+import bookingTypes from '../../common/booking/bookingTypes';
+import bookingStatus from '../../common/booking/bookingStatuses';
+import type { MultictyBooking_booking } from './__generated__/MulticityBooking_booking.graphql';
+
+type Props = {
+  booking: MultictyBooking_booking,
+};
+
+const MulticityBooking = (props: Props) => {
+  const booking = props.booking;
+  const { databaseId, passengerCount } = booking;
+  const status = booking.status && bookingStatus[booking.status];
+
+  const trip = idx(booking, _ => _.trips[0]) || [];
+  const departureDate = idx(booking, _ => _.start.time) || '';
+  const origin = idx(booking, _ => _.start.airport.city.name) || '';
+  const IATAOrigin = idx(booking, _ => _.start.airport.locationId) || '';
+  const destination = idx(booking, _ => _.end.airport.city.name) || '';
+  const IATADestination = idx(booking, _ => _.end.airport.locationId) || '';
+
+  return (
+    <BookingCard
```

Comment:
`BookingCard` component is pretty long with quite a lot of properties, isn't it? I would suggest to leverage Relay fragments and decompose it a little bit. It's absolutely fine to compose small components with tiny fragments. In fact we have one rule - write fragments only for what your components need. You are fetching here data for `BookingCard` which we consider to be a wrong approach... 🙂

Reply:
BTW: this is how we do it: https://github.com/kiwicom/mobile/blob/5074b23d8c0b59c6a9b7d3514b5eac7b9a4c09f1/app/MMB/src/list/MulticityFlight.js#L28

As you can see we are not even fetching real data there - only spreading the fragments in order to deal with the multiple booking types we have. This greatly simplifies our code because we don't have to repeat ourselves (see how the fragments are sooo similar here)...